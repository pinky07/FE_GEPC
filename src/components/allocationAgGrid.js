import React from 'react';
import { connect } from 'react-redux';
import {
  Row,
  Col,
} from 'reactstrap';
import {AgGridReact} from "ag-grid-react";
import allocationModel from '../model/allocationModel';
import MixDetails from './mixDetails';
import MixStatistics from './mixStatistics';

let columnDefs = [
  {
    field: 'accountgroupname',
    headerName: 'Node Name',
    width: 350,
  },
  {
    field: 'assetCat',
    headerName: 'Asset Category',
    width: 150,
  },
  {
    field: 'newGrouping',
    headerName: 'New Grouping',
    width: 150,
  },
  {
    field: 'aamb',
    headerName: 'AAMB',
    width: 150,
  },
  {
    field: 'nd',
    headerName: 'Node',
    width: 150,
  },
  {
    field: 'alias',
    headerName: 'Asset Class Alias',
    width: 150,
  },
  {
    field: 'policy',
    headerName: 'Policy',
    width: 150,
  },
  {
    field: 'value',
    headerName: 'Value',
    width: 150,
  },
  {
    field: 'mixA',
    headerName: 'Mix A',
    width: 150,
    editable: true,
  },
  {
    field: 'mixB',
    headerName: 'Mix B',
    width: 150,
    editable: true,
  },
  {
    field: 'mixC',
    headerName: 'Mix C',
    width: 150,
    editable: true,
  }
];

class Aggrid extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      columnDefs
    }
  }
  componentWillReceiveProps (nextProps) {
    if (nextProps.gridData !== this.props.gridData) {
      allocationModel().getAllocationGrid(nextProps.gridData).then(rows => { console.log(rows)
        this.setState({
          rows,
        });
      });
    }
  }

  onGridReady(params) {
    this.gridApi = params.api;
    this.columnApi = params.columnApi;
    this.gridApi.sizeColumnsToFit();
  }

  render () {

    return (
      <div className="allocationGrid">
        <Row>
          <Col lg="9" md="9" className="gridContainer">
            <div className="aggridcontainer ag-fresh">
              <AgGridReact
                columnDefs={this.state.columnDefs}
                rowData={this.state && this.state.rows}
                onGridReady={this.onGridReady}
                headerHeight="35"
              />
            </div>
          </Col>
          <Col lg="3" md="3">
            <MixStatistics />
            <MixDetails />
          </Col>
        </Row>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    ...state.allocationGrid
  };
};

export default connect(mapStateToProps, null)(Aggrid);
