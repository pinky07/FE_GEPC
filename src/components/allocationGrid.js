import React from 'react';
import { connect } from 'react-redux';
import createReactClass from 'create-react-class';
import {
  Row,
  Col,
} from 'reactstrap';
import ReactDataGrid from 'react-data-grid';
import update from 'immutability-helper';
import RowRenderer from './rowRenderer';
import allocationModel from '../model/allocationModel';
import ExpanderFormatter from './colNameFormatter';
import MixDetails from './mixDetails';
import MixStatistics from './mixStatistics';
import BetaGroupDropdown from './betaGroupDropdown';

let columns = [
  {
    key: 'node',
    name: 'Node Name',
    width: 350,
    formatter: ExpanderFormatter
  },
  {
    key: 'assetCat',
    name: 'Asset Category',
    width: 150,
  },
  {
    key: 'newGrouping',
    name: 'New Grouping',
    width: 150,
    editable: true
  },
  {
    key: 'aamb',
    name: 'AAMB',
    width: 150,
  },
  {
    key: 'nd',
    name: 'Node',
    width: 150,
  },
  {
    key: 'alias',
    name: 'Asset Class Alias',
    width: 150,
  },
  {
    key: 'policy',
    name: 'Policy',
    width: 100,
  },
  {
    key: 'value',
    name: 'Value',
    width: 100,
  },
  {
    key: 'mixA',
    name: 'Mix A',
    width: 100,
  }
];

const AllocationGrid = createReactClass({
  getInitialState() {
    return { expanded: {}, rows: [] };
  },

  getRows(i) {
    return this.state.rows[i];
  },

  handleGridRowsUpdated({ fromRow, toRow, updated }) {
    let rows = this.state.rows.slice();

    for (let i = fromRow; i <= toRow; i++) {
      let rowToUpdate = rows[i];
      let updatedRow = update(rowToUpdate, {$merge: updated});
      rows[i] = updatedRow;
    }

    this.setState({ rows });
  },

  componentWillReceiveProps (nextProps) {
    if (nextProps.gridData !== this.props.gridData) {
      allocationModel().getAllocationGrid(nextProps.gridData).then(rows => {
        this.setState({
          rows,
        });
      });
    }
  },

  render() {
    return (
      <div className="allocationGrid">
        <Row>
          <Col lg="12" md="12" className="betagroupDropdown">
            <BetaGroupDropdown/>
          </Col>
        </Row>
        <Row>
          <Col lg="9" md="9" className="gridContainer">
            <ReactDataGrid
              enableCellSelect={true}
              columns={columns}
              rowGetter={this.getRows}
              rowsCount={this.state.rows.length}
              minHeight={500}
              rowRenderer={RowRenderer}
              onGridRowsUpdated={this.handleGridRowsUpdated}
            />
          </Col>
          <Col lg="3" md="3">
            <MixStatistics />
            <MixDetails />
          </Col>
        </Row>
      </div>
    );
  }
});

const mapStateToProps = state => {
  return {
    ...state.allocationGrid
  };
};

export default connect(mapStateToProps, null)(AllocationGrid);
