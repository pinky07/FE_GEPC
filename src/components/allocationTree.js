import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { ContextMenuProvider } from 'react-contexify';
import {
  Row,
  Col,
  Input,
} from 'reactstrap';
import SortableTree from 'react-sortable-tree';
import {
  getAllocationTree,
  saveAllocationTree,
  selectNode,
  updateTree,
} from '../actions';
import Constants from '../services/constants';
import NodeMenu from './nodeMenu';
import TreeButtonBar from './treeButtonBar';
import NodeDetails from './nodeDetails';
import BetaGroupDropdown from './betaGroupDropdown';
import TreeNodeRenderer from './treeNodeRenderer';

export class AllocationTree extends React.Component {

  componentDidMount () {
    this.props.getAllocationTree();
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.tree !== this.props.tree) {
      if (!this.props.selectedNode) {
        this.props.selectNode({ node: nextProps.tree.data[0] });
      }
    }
  }

  saveTree = () => {
    this.props.saveAllocationTree(this.props.tree);
  }

  treeButtons = rowInfo => {
    const buttons = [
      <ContextMenuProvider id="menu_id" event="onClick">
        <button
          onClick={() => this.props.selectNode(rowInfo)}
        >
          ...
        </button>
      </ContextMenuProvider>,
      <button
        className="buttonInfoNode"
        onClick={() => this.props.selectNode(rowInfo)}
      >
        ...
      </button>,
    ];

    return buttons;
  }

  render () {
    const { tree } = this.props;
    return (
      <div className="allocationTree">
        <Row className="treeHeader">
          <Col lg="3" md="3" sm="12" xs="12">
            <h5>{tree.name}</h5>
          </Col>

          <Col lg="5" md="5" sm="12" xs="12">
            <TreeButtonBar
              save={this.saveTree}
            />
          </Col>
          <Col lg="4" md="4" sm="12" xs="12">
            <div className="settings">
              <Input placeholder="Global Set" />
              <BetaGroupDropdown />
            </div>
          </Col>
        </Row>

        <Row>
          <Col lg="8" md="8" sm="12" className="treecontainer">
            <SortableTree
              treeData={tree.data}
              onChange={treeData => this.props.updateTree(treeData)}
              canDrop={ ({ nextPath }) => nextPath.length > 1}
              maxDepth={Constants.MAX_TREE_DEPTH}
              generateNodeProps={rowInfo => ({
                buttons: this.treeButtons(rowInfo),
              })}
              nodeContentRenderer={TreeNodeRenderer}
            />
            <NodeMenu/>
          </Col>
          <Col lg="4" md="4" sm="12">
            <NodeDetails />
          </Col>
        </Row>
      </div>
    );
  }
}

AllocationTree.propTypes = {
  tree: PropTypes.object.isRequired,
};

AllocationTree.defaultProps = {
  tree: {
    name: '',
    data: []
  },
  treeData: [],
};

const mapStateToProps = state => {
  return {
    ...state.allocationTree
  };
};

export default connect( mapStateToProps, {
  getAllocationTree,
  saveAllocationTree,
  selectNode,
  updateTree,
} )(AllocationTree);
