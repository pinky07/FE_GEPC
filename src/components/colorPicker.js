import React from 'react'
import reactCSS from 'reactcss'
import { SketchPicker } from 'react-color'

export default class ColorPicker extends React.Component {

  constructor (props) {
    super(props);
    this.state = {
      displayColorPicker: false,
      color: {
        r: '0',
        g: '32',
        b: '96',
        a: '1',
      },
    };
  }

  handleClick = () => {
    this.setState({ displayColorPicker: !this.state.displayColorPicker })
  };

  handleClose = () => {
    this.setState({ displayColorPicker: false })
  };

  handleChange = (color) => {
    this.setState({ color: color.rgb })
  };

  renderPicker () {
    if (this.state.displayColorPicker) {
      return (
        <div className="popover">
          <div className="cover" onClick={ this.handleClose }/>
          <SketchPicker color={ this.state.color } onChange={ this.handleChange } />
        </div>
      );
    }
    return null;
  }

  render() {

    const styles = reactCSS({
      'default': {
        color: {
          background: `rgba(${ this.state.color.r }, ${ this.state.color.g }, ${ this.state.color.b }, ${ this.state.color.a })`,
        },
      },
    });

    return (
      <div className="colorPicker">
        <div onClick={ this.handleClick }>
          <div className="color" style={ styles.color } />
        </div>
        { this.renderPicker() }
      </div>
    )
  }
}
