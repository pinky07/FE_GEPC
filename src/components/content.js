import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  Row,
  Col,
  Container,
  TabContent,
  TabPane,
  Nav,
  NavItem,
  NavLink
} from 'reactstrap';
import TreeView from './treeView/treeView';
import GridView from './gridView/gridView';

const TREE_VIEW_TAB = '1';
const GRID_VIEW_TAB = '2';

/**
 * Content component with Grid and Tree View tabs.
 * 
 * @author Francisco ZuÃ±iga
 * @class Content
 * @extends {React.Component}
 */
export class Content extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      activeTab: TREE_VIEW_TAB,
    };
  }

  toggle = index => {
    if (this.state.activeTab !== index) {
      this.setState({
        activeTab: index
      });
    }
  }

  render() {
    return (
      <Container className="content" fluid={true}>
        {this.props.isLoading ? <div className="loader"></div> : null}
        <Row>
          <Col lg="12" md="12">
            <Nav tabs>
              <NavItem>
                <NavLink
                  className={this.state.activeTab === TREE_VIEW_TAB ? 'active' : ''}
                  onClick={() => { this.toggle(TREE_VIEW_TAB); }}
                >
                  Tree View
                </NavLink>
              </NavItem>

              <NavItem>
                <NavLink
                  className={this.state.activeTab === GRID_VIEW_TAB ? 'active' : ''}
                  onClick={() => { this.toggle(GRID_VIEW_TAB); }}
                >
                  Grid View
                </NavLink>
              </NavItem>
            </Nav>
            <TabContent activeTab={this.state.activeTab}>
              <TabPane tabId={TREE_VIEW_TAB}>
                <Row>
                  <Col lg="12">
                    <TreeView />
                  </Col>
                </Row>
              </TabPane>

              <TabPane tabId={GRID_VIEW_TAB}>
                <Row>
                  <Col lg="12">
                    <GridView />
                  </Col>
                </Row>
              </TabPane>
            </TabContent>

          </Col>
        </Row>
      </Container>
    );
  }
}

Content.propTypes = {
  isLoading: PropTypes.bool.isRequired,
};

Content.defaultProps = {
  isLoading: false
};

const mapStateToProps = state => {
  return {
    ...state.shared
  };
};

export default connect(mapStateToProps, null)(Content);
