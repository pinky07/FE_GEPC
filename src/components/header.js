import React from 'react';
import { connect } from 'react-redux';
import {
  Button,
  Col,
  Container,
  Row,
} from 'reactstrap';

import {
  save,
} from '../actions';

/**
 * Header component with Segment, Client and Plan.
 * 
 * @author Francisco Zuñiga
 * @author Rubén Jiménez
 * @class Header
 * @extends {React.Component}
 */
export class Header extends React.Component {

  /**
   * Saves everything displayed in the application to the backend by dispatching an action.
   * 
   * @memberof Header
   */
  save = event => {
    console.log("Save button clicked!");
    this.props.save();
  }

  /**
   * Displays help information.
   * 
   * @memberof Header
   */
  help = event => {
    console.log("Help button clicked!");
  }

  /**
   * Renders this component
   * 
   * @returns Rendered HTML
   * @memberof Header
   */
  render() {
    const { segment } = this.props
    const { client } = this.props
    const { plan } = this.props

    return (
      <div className="app-header">
        <Container fluid>
          <Row>
            <Col xs="12" md="2" className="vcenter-parent text-left">
              <p className="vcenter">NEPC Tree View</p>
            </Col>
            <Col xs="12" md="8" className="vcenter-parent text-center">
              <p className="vcenter">Segment: <span>{segment}</span> Client: <span>{client}</span> Plan: <span>{plan}</span></p>
            </Col>
            <Col xs="12" md="2" className="text-right">
              <Button color="primary" className="btn-save" onClick={this.save}>Save</Button>
              <Button color="primary" className="btn-help" onClick={this.help}>Help</Button>
            </Col>
          </Row>
        </Container>
      </div>
    );
  }
}

/**
 * Maps relevant properties to this component from the store.
 * 
 * @param {*} state 
 */
const mapStateToProps = state => {
  return {
    ...state.spotFire
  };
};

/**
 * Connects this component to Redux
 */
export default connect(
  mapStateToProps,
  {
    save,
  }
)(Header);