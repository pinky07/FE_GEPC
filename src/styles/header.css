.dropdown {
 display: inline;
 margin-left: 1em;
}