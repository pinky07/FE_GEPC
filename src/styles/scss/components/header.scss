.app-header {
  background-color: $blue-1;
  padding: 20px;

  .btn-save, .btn-help {
    background: $green;
  }

  .btn-save {
    margin-right: 10px;
  }

  p {
    margin: 0;
    color: white;
    font-weight: bold;
    font-family: $font;
    vertical-align: middle;
  }
}
