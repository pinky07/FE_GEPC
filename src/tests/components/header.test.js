import React from 'react';
import { mount  } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Header } from '../../components/header';

let props = {
  segment: 'Segment',
  client: 'Client',
  plan: 'Plan',
  save: jest.fn(),
};

describe('header component', () => {
  let wrapper = mount(<Header {...props}/>);

  it ('renders correctly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});