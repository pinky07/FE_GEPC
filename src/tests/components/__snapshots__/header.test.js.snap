// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component renders correctly 1`] = `
<Header
  client="Client"
  plan="Plan"
  save={[Function]}
  segment="Segment"
>
  <div
    className="app-header"
  >
    <Container
      fluid={true}
      tag="div"
    >
      <div
        className="container-fluid"
      >
        <Row
          tag="div"
        >
          <div
            className="row"
          >
            <Col
              className="vcenter-parent text-left"
              md="2"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="12"
            >
              <div
                className="vcenter-parent text-left col-12 col-md-2"
              >
                <p
                  className="vcenter"
                >
                  NEPC Tree View
                </p>
              </div>
            </Col>
            <Col
              className="vcenter-parent text-center"
              md="8"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="12"
            >
              <div
                className="vcenter-parent text-center col-12 col-md-8"
              >
                <p
                  className="vcenter"
                >
                  Segment: 
                  <span>
                    Segment
                  </span>
                   Client: 
                  <span>
                    Client
                  </span>
                   Plan: 
                  <span>
                    Plan
                  </span>
                </p>
              </div>
            </Col>
            <Col
              className="text-right"
              md="2"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="12"
            >
              <div
                className="text-right col-12 col-md-2"
              >
                <Button
                  className="btn-save"
                  color="primary"
                  onClick={[Function]}
                  tag="button"
                >
                  <button
                    className="btn-save btn btn-primary"
                    onClick={[Function]}
                    type="button"
                  >
                    Save
                  </button>
                </Button>
                <Button
                  className="btn-help"
                  color="primary"
                  onClick={[Function]}
                  tag="button"
                >
                  <button
                    className="btn-help btn btn-primary"
                    onClick={[Function]}
                    type="button"
                  >
                    Help
                  </button>
                </Button>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Container>
  </div>
</Header>
`;
